test/built-ins/Iterator/prototype/Symbol.toStringTag/weird-setter.js

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2023 Michael Ficarra. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-set-iteratorprototype-@@tostringtag
+description: weird setter
+info: |
+  The value of the [[Get]] attribute is a built-in function that requires no arguments. It performs the following steps when called:
+    1. Return *"Iterator"*.
+
+  The value of the [[Set]] attribute is a built-in function that takes an argument _v_. It performs the following steps when called:
+    1. Perform ? SetterThatIgnoresPrototypeProperties(%Iterator.prototype%, %Symbol.toStringTag%, _v_).
+    2. Return *undefined*.
+features: [iterator-helpers]
+---*/
+
+let IteratorPrototype = Object.getPrototypeOf(Object.getPrototypeOf([][Symbol.iterator]()))
+let GeneratorPrototype = Object.getPrototypeOf(function*(){}.prototype);
+
+let sentinel = 'a';
+
+let { get, set } = Object.getOwnPropertyDescriptor(Iterator.prototype, Symbol.toStringTag);
+
+assert.sameValue(Iterator.prototype[Symbol.toStringTag], 'Iterator');
+assert.sameValue(get.call(), 'Iterator');
+
+// 1. If _this_ is not an Object, then
+//   1. Throw a *TypeError* exception.
+assert.throws(() => set.call(undefined, ''));
+assert.throws(() => set.call(null, ''));
+assert.throws(() => set.call(true, ''));
+
+// 1. If _this_ is _home_, then
+//   1. NOTE: Throwing here emulates assignment to a non-writable data property on the _home_ object in strict mode code.
+//   1. Throw a *TypeError* exception.
+assert.throws(() => set.call(IteratorPrototype, ''));
+assert.throws(() => IteratorPrototype[Symbol.toStringTag] = '');
+
+assert.sameValue(Iterator.prototype[Symbol.toStringTag], 'Iterator');
+assert.sameValue(get.call(), 'Iterator');
+
+// 1. If _desc_ is *undefined*, then
+//   1. Perform ? CreateDataPropertyOrThrow(_this_, _p_, _v_).
+let o = {};
+set.call(o, sentinel);
+assert.sameValue(o[Symbol.toStringTag], sentinel);
+
+assert.sameValue(Iterator.prototype[Symbol.toStringTag], 'Iterator');
+assert.sameValue(get.call(), 'Iterator');
+
+// 1. Else,
+//   1. Perform ? Set(_this_, _p_, _v_, *true*).
+Object.freeze(IteratorPrototype);
+GeneratorPrototype[Symbol.toStringTag] = sentinel;
+assert.sameValue(GeneratorPrototype[Symbol.toStringTag], sentinel);
```

Comment:
`GeneratorPrototype`'s [[Prototype]] has a non-writable own data property named `Symbol.toStringTag`, so this line seems wrong.

h/t @rmahdav